src/Controller/Security/ResetPasswordController.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace App\Controller\Security;
+
+use App\Entity\User;
+use App\Form\ChangePasswordForm;
+use App\Form\ForgotPasswordForm;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\DependencyInjection\Attribute\Target;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\UriSigner;
+use Symfony\Component\Mailer\Header\MetadataHeader;
+use Symfony\Component\Mailer\Header\TagHeader;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\RateLimiter\RateLimiterFactory;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+use function Symfony\Component\Clock\now;
+
+final class ResetPasswordController extends AbstractController
+{
+    private const EXPIRES_AFTER = 3600; // 1 hour
+    private const SESSION_ID = 'password_reset_id';
+    private const ALREADY_USED_HASH_PARAM = 'valid';
+
+    public function __construct(
+        private UriSigner $uriSigner,
+        private MailerInterface $mailer,
+        private UserRepository $users,
+        private UserPasswordHasherInterface $passwordHasher,
+        private EntityManagerInterface $em,
+        private Security $security,
```

Comment:
I'm a bit partial to method injection. This looks more technical & prevents laziness (e.g. I don't need the mailer for most actions). I can see why you did this (so you can access these from any method), but I'm curious how things look/feel with method injection for at least most services (passing the services to private methods manually)